### PR TITLE
[MODE-2694] Close input streams passed to TikaNameOnlyDetector.detect()

### PR DESCRIPTION
**Pull request by dalbani**
https://issues.jboss.org/browse/MODE-2694

### CODE REVIEW

File: modeshape-jcr/src/main/java/org/modeshape/jcr/mimetype/tika/TikaNameOnlyDetector.java [hunk 1]
```diff
@@ -61,7 +63,8 @@ protected void initDetector( ClassLoader loader ) {
     @Override
     protected String detect( InputStream inputStream, Metadata metadata ) {
         try {
-            // we never care about the stream here
+            // we never care about the stream here, but close it anyway
+            IoUtil.closeQuietly(inputStream);
```
Comment (hchiorean) [inline modeshape-jcr/src/main/java/org/modeshape/jcr/mimetype/tika/TikaNameOnlyDetector.java]: I would move this inside a `finally` block in the same method

Comment (dalbani) [inline modeshape-jcr/src/main/java/org/modeshape/jcr/mimetype/tika/TikaNameOnlyDetector.java]: Yep, agree.

Comment (dalbani) [inline modeshape-jcr/src/main/java/org/modeshape/jcr/mimetype/tika/TikaNameOnlyDetector.java]: I happen to get this kind of S3-related warnings in relation to this unused stream:
```Not all bytes were read from the S3ObjectInputStream, aborting HTTP connection. This is likely an error and may result in sub-optimal behavior. Request only the bytes you need via a ranged GET or drain the input stream after use.```

Comes from https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/S3AbortableInputStream.java#L159.

**Comment by hchiorean** [comment]
@dalbani this looks ok, with the exception of the previous comment - i.e. moving the logic of closing the stream inside a `finally` block

**Comment by dalbani** [comment]
Moved into `finally` block.

**Pull request merged**